packages/lib-core/src/store/PluginStoreContext.tsx

```diff
@@ -31,3 +31,12 @@ export const usePluginStore = (): PluginConsumer & PluginManager => {
 
   return store;
 };
+
+/**
+ * React hook that provides access to the feature flags.
+ */
+export const useFeatureFlags = (): [FeatureFlags, (newFeatureFlags: FeatureFlags) => void] => {
+  const store = usePluginStore();
+
+  return [store.getFeatureFlags(), store.setFeatureFlags];
+};
```

Comment:
Often within a Component we're only looking at a single feature flag.
An API that simplifies this, similar to `useState` may be best?

```
const [flag, setFlag] = useFeatureFlag('MY_FLAG');
...
setFlag(true);
```

vs
```
const [flags, setFeatureFlag] = useFeatureFlags();
const flag = flags['MY_FLAG'];
...
setFeatureFlags({ 'MY_FLAG': true });
```

WDYT?
cc @vojtechszocs

Reply:
@christianvogt That's a good point. If it's more common for a single flag to be manipulated at a time, I agree that useFeatureFlag makes more sense. Do you foresee apps needing access to all the available feature flags at some point? in which case we could probably keep both hooks - for manipulating a single flag as well as multiple flags?
Let me know what you think @christianvogt @vojtechszocs. Thanks!

Reply:
Flag usage is very often specific to individual checks. I don't see the need for a single component wanting access to all flags at once. There will be instances where more than one flag is checked at a time, in which case multiple uses of the hook will be sufficient.